tests/unit/configFile.spec.js

```diff
@@ -133,4 +133,239 @@ describe("configFile.js", () => {
+    });
+
+    it("queries are overridden or ignored depending on corresponding flag", async () => {
+      var newRSA = _.cloneDeep(defaultSettings.RSA);
+      newRSA.Queries = [
+        [-1, "Search Destination IP", "ip.dst=TESTSEARCH", true],
+        [-1, "New Search Destination IP", "new_ip.dst=TESTSEARCH", true],
+      ];
+
+      // Should override queries if flag is override.
+      var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      settings.mergeRSA.queries = "override";
+      await LocalStore.setOne(StoreKey.SETTINGS, settings);
+      await ConfigFile.default.updateSpecialProvider(
+        StoreKey.RSA_SECURITY,
+        newRSA,
+        "mergeRSA"
+      );
+      var result = await LocalStore.getOne(StoreKey.RSA_SECURITY);
+      var expectedResult = ConfigFile.default.parseQueries([
+        [-1, "Search Destination IP", "ip.dst=TESTSEARCH", true],
+        [-1, "New Search Destination IP", "new_ip.dst=TESTSEARCH", true],
+      ]);
+      expect(result.queries).toEqual(expectedResult);
+
+      // Should ignore queries if flag is ignore.
+      newRSA.Queries = [
+        [-1, "Search Hostname", "alias.host='TESTSEARCH'", true],
+      ];
+      settings.mergeRSA.queries = "ignore";
+      await LocalStore.setOne(StoreKey.SETTINGS, settings);
+      await ConfigFile.default.updateSpecialProvider(
+        StoreKey.RSA_SECURITY,
+        newRSA,
+        "mergeRSA"
+      );
+      result = await LocalStore.getOne(StoreKey.RSA_SECURITY);
+      expect(result.queries).toEqual(expectedResult);
+    });
+  });
+
+  describe("parseJSONFile function", () => {
+    it("basic settings should be overridden if the corresponding flag is enabled", async () => {
+      await LocalStore.clear();
+      const expectedParsedData = {
+        configurationURL:
+          "https://raw.githubusercontent.com/AdvancedThreatAnalytics/threat-analytics-search/main/settings.json",
+        useGroups: true,
+        configEncrypted: false,
+        configEncryptionKey: "",
+        autoUpdateConfig: false,
+      };
+
+      // Shouldn't override if the flag is false.
+      await ConfigFile.default.parseJSONFile(defaultSettings, false);
+      var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      expect(settings).toEqual({});
+
+      // Should override if the flag is true.
+      await ConfigFile.default.parseJSONFile(defaultSettings, true);
+      settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      expect(settings).toEqual(expect.objectContaining(expectedParsedData));
+    });
+
+    it("groups name should be updated only if the mergeGroups is true", async () => {
+      var newData = _.cloneDeep(defaultSettings);
+      var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      const defaultProvidersGroup = [
+        { name: "IP Lookup", enabled: true },
+        { name: "Domain", enabled: true },
+        { name: "Hash", enabled: false },
+      ];
```

Comment:
Same as above: this test will break if we modify any value on `settings.json`, instead you could do `const defaultGroups = ConfigFile.parseGroups(defaultSettings.groups);`